anchore_engine/services/apiext/api/helpers/image_content_response.py

```diff
@@ -0,0 +1,220 @@
+            el["type"] = "GEM"
+            el["location"] = package
+            el["version"] = content_data[package]["versions"][0]
+            el["origin"] = ",".join(content_data[package]["origins"]) or "Unknown"
+            el["license"] = " ".join(content_data[package]["lics"]) or "Unknown"
+            el["licenses"] = content_data[package]["lics"] or ["Unknown"]
+        except:
+            continue
+        response.append(el)
+    return response
+
+
+def _build_python_response(content_data):
+    response = []
+    for package in list(content_data.keys()):
+        el = {}
+        try:
+            el["package"] = content_data[package]["name"]
+            el["type"] = "PYTHON"
+            el["location"] = content_data[package]["location"]
+            el["version"] = content_data[package]["version"]
+            el["origin"] = content_data[package]["origin"] or "Unknown"
+            el["license"] = content_data[package]["license"] or "Unknown"
+            el["licenses"] = content_data[package]["license"].split(" ") or ["Unknown"]
+        except:
+            continue
+        response.append(el)
+    return response
+
+
+def _build_java_response(content_data):
+    response = []
+    for package in list(content_data.keys()):
+        el = {}
+        try:
+            el["package"] = content_data[package]["name"]
+            el["type"] = content_data[package]["type"].upper()
+            el["location"] = content_data[package]["location"]
+            el["specification-version"] = content_data[package]["specification-version"]
+            el["implementation-version"] = content_data[package]["implementation-version"]
+            el["maven-version"] = content_data[package]["maven-version"]
+            el["origin"] = content_data[package]["origin"] or "Unknown"
+        except:
+            continue
+        response.append(el)
+    return response
+
+
+def _build_files_response(content_data):
+    response = []
+    elmap = {
+        "linkdst": "linkdest",
+        "size": "size",
+        "mode": "mode",
+        "sha256": "sha256",
+        "type": "type",
+        "uid": "uid",
+        "gid": "gid",
+    }
+    for filename in list(content_data.keys()):
+        el = {}
+        try:
+            el["filename"] = filename
+            for elkey in list(elmap.keys()):
+                try:
+                    el[elmap[elkey]] = content_data[filename][elkey]
+                except:
+                    el[elmap[elkey]] = None
+
+            # special formatting
+            el["mode"] = format(stat.S_IMODE(el["mode"]), "05o")
+            if el["sha256"] == "DIRECTORY_OR_OTHER":
+                el["sha256"] = None
+        except:
+            continue
+        response.append(el)
+    return response
+
+
+def _build_docker_history_response(content_data):
+    try:
+        return utils.ensure_str(
+            base64.encodebytes(utils.ensure_bytes(json.dumps(content_data)))
```

Comment:
This encoding pattern is repeated a few times, could this be extracted into a helper that accepts the incoming string? In this case `content_data` gets dumped, but the use is the same on the next three

Reply:
@alfredodeza see the updated commit; these were slightly different actions (one required `json.dumps`) so a lambda was used to encapsulate providing data within the try-except block.